onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1156,7 +1160,30 @@ std::unique_ptr<IExecutionProvider> CreateExecutionProviderInstance(
 #if !defined(__APPLE__)
     LOGS_DEFAULT(WARNING) << "CoreML execution provider can only be used to generate ORT format model in this build.";
 #endif
-    return onnxruntime::CoreMLProviderFactoryCreator::Create(0)->CreateProvider();
+    uint32_t coreml_flags = 0;
+
+    const auto it = provider_options_map.find(type);
+    if (it != provider_options_map.end()) {
+      const ProviderOptions& options = it->second;
+      auto flags = options.find("flags");
+      if (flags != options.end()) {
+        const auto& flags_str = flags->second;
+
+        if (flags_str.find("COREML_FLAG_USE_CPU_ONLY") != std::string::npos) {
```

Comment:
would it be worth enabling the string key/value configuration for the CoreML EP and making it consistent across Python and the `SessionOptionsAppendExecutionProvider` C API?

this is fine as a near-term implementation but eventually it might be good to have a consistent way to configure it. then we can also point to a single place that documents it.

Reply:
Yeah - would be good to use that setup and make it possible to enable with the generic SessionOptionsAppendExecutionProvider.

That should extend to exposing relevant options to onnx_test_runner and onnxruntime_perf_test. Based on recent experience at least enabling ML Program and specifying CPU_ONLY are needed. 

I'll add a task to the backlog.

Reply:
Add Task 40739